HealthcareTA/HealthcareTA.cs

```diff
@@ -0,0 +1,136 @@
+                    }
+                    List<string> batchInput = new List<string>()
+                    {
+                        document
+                    };
+
+                    // start analysis process
+                    var timer = System.Diagnostics.Stopwatch.StartNew();
+                    AnalyzeHealthcareEntitiesOperation healthOperation = await client.StartAnalyzeHealthcareEntitiesAsync(batchInput, "en", options);
+                    var healthOperationTask = healthOperation.WaitForCompletionAsync().AsTask();
+
+                    if (await Task.WhenAny(healthOperationTask, Task.Delay(timeoutMiliseconds)) == healthOperationTask)
+                    {
+                        // Task Completed, now lets process the result.
+                        outRecord.Data["status"] = healthOperation.Status.ToString();
+                        if (healthOperation.Status != TextAnalyticsOperationStatus.Succeeded)
+                        {
+                            // The operation was not a success
+                            outRecord.Warnings.Add(new WebApiErrorWarningContract
+                            {
+                                Message = "Healthcare Text Analytics Error: Health Operation returned a non-succeeded status."
+                            });
+                        }
+                        else
+                        {
+                            // The operation was a success, so lets add the results to our output.
+                            await ExtractEntityData(healthOperation.Value, outRecord);
+                        }
+                    }
+                    else
+                    {
+                        // Timeout
+                        outRecord.Warnings.Add(new WebApiErrorWarningContract
+                        {
+                            Message = "Healthcare Text Analytics Error: The Text Analysis Operation took too long to complete."
+                        });
+                    }
+
+                    // Record how long this task took to complete.
+                    timer.Stop();
+                    outRecord.Data["timeToComplete"] = timer.Elapsed.TotalSeconds.ToString();
+
+                    return outRecord;
+                });
+
+            return new OkObjectResult(response);
+        }
+
+        private static async Task ExtractEntityData(AsyncPageable<AnalyzeHealthcareEntitiesResultCollection> pages, WebApiResponseRecord outRecord)
+        {
+            await foreach (AnalyzeHealthcareEntitiesResultCollection documentsInPage in pages)
+            {
+                foreach (AnalyzeHealthcareEntitiesResult entitiesInDoc in documentsInPage)
+                {
+                    if (!entitiesInDoc.HasError)
+                    {
+                        outRecord.Data["entities"] = entitiesInDoc.Entities;
+                        outRecord.Data["relations"] = entitiesInDoc.EntityRelations;
```

Comment:
Aren't you getting data from only the last entities on the last page?

Reply:
Good catch, he's overwriting instead of appending each time. @jonathanserbent make sure to fix

Reply:
Just started looking into this, I need to read up more on the docs for the TA library, but currently the results I have checked all have just one document (which makes sense because for each request we are only sending one), but it also only ever has one collection of entities/entity relations.

Reply:
Went over this with Jonathan, the TA API doesn't make that much sense but we assume they made it pageable in the SDK just in case they make the actual API pageable in the future. With Jonathan's current non-batching implementation this should work as is, but I did ask him to just make sure we create a list that we append to just in case.

Reply:
Implemented and pushed the change Carey Mentioned above.